Paper: Cell Tracking in 3D using deep learning segmentations

kapoorlab opened the pull request:
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@stargaser<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@NickleDave, @davekits<!--end-reviewers-list-->

kapoorlab committed:
first commit

kapoorlab committed:
adding requirements files

kapoorlab committed:
paper first commit

kapoorlab committed:
little introduction

kapoorlab committed:
little equation

kapoorlab committed:
little equation

kapoorlab committed:
little equation

kapoorlab committed:
little equation

kapoorlab committed:
little equation

kapoorlab committed:
little equation

kapoorlab committed:
add reference

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
add reference

kapoorlab committed:
add reference

kapoorlab committed:
add reference

kapoorlab committed:
add reference

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
linux transfer

kapoorlab committed:
napari figures

kapoorlab committed:
tracking

kapoorlab committed:
add bib

kapoorlab committed:
turn off comp

kapoorlab committed:
turn off comp

kapoorlab committed:
Delete papers/varun_kapoor/old_figs directory

kapoorlab committed:
first few sections

kapoorlab committed:
Merge branch '2021' of https://github.com/kapoorlab/scipy_proceedings into 2021

kapoorlab committed:
removing bull

kapoorlab committed:
speel check

kapoorlab committed:
draft ready

kapoorlab committed:
denoise both

kapoorlab committed:
changing introduction

kapoorlab committed:
changing introduction

kapoorlab committed:
include bib

kapoorlab committed:
ref invisible

kapoorlab committed:
intro

kapoorlab committed:
intro

kapoorlab committed:
intro

kapoorlab committed:
methods

kapoorlab committed:
metrics

kapoorlab committed:
metrics

kapoorlab committed:
metrics

kapoorlab committed:
metrics

kapoorlab committed:
metrics

kapoorlab committed:
figures

kapoorlab committed:
figures

kapoorlab committed:
figures

kapoorlab committed:
figures

kapoorlab committed:
citations

kapoorlab committed:
references

kapoorlab committed:
figures

kapoorlab committed:
draft

kapoorlab committed:
draft

kapoorlab committed:
name char

kapoorlab committed:
methods

kapoorlab committed:
methods

kapoorlab committed:
bullet points

kapoorlab committed:
bullet points

kapoorlab committed:
bullet points

kapoorlab committed:
bullet points

kapoorlab committed:
update figure legend

kapoorlab committed:
update figure legend

kapoorlab committed:
update figure legend

kapoorlab committed:
conclusions

kapoorlab committed:
conclusions

kapoorlab committed:
conclusions

kapoorlab committed:
conclusions

kapoorlab committed:
conclusions

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
citation tr

kapoorlab committed:
Update 00_varunclaudia.rst

kapoorlab committed:
rst

kapoorlab committed:
guig changes

kapoorlab committed:
guig changes

kapoorlab committed:
guig changes

kapoorlab committed:
guig changes

kapoorlab committed:
guig changes

kapoorlab committed:
france funding

kapoorlab committed:
requirements

kapoorlab committed:
other papers

kapoorlab committed:
final version

scoobies commented:
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

Commit:
Merge branch '2021' into 2021

deniederhut commented:
@scoobies assign @stargaser as editor

scoobies commented:
Assigned! @stargaser is now the editor

deniederhut commented:
@scoobies add @NickleDave to reviewers

scoobies commented:
@NickleDave added to the reviewers list!

deniederhut commented:
We've been having difficulty finding reviewers for some papers, so to make sure everyone has enough time to do their reviews, we will be bumping the deadline for the initial review back to June 17th.

Commit:
Merge branch '2021' into 2021

stargaser commented:
scoobies add @davekits to reviewers

stargaser commented:
@scoobies add @davekits to reviewers

scoobies commented:
@davekits added to the reviewers list!

kapoorlab committed:
code update

kapoorlab committed:
paper eq

kapoorlab commented:
@scoobies help

scoobies commented:
Hello @kapoorlab, here are the things you can ask me to do:

```

# List all available commands
@scoobies help

# Show our community Code of Conduct and Guidelines
@scoobies code of conduct

# Add to this issue's reviewers list
@scoobies add @username to reviewers

# Remove from this issue's reviewers list
@scoobies remove @username from reviewers

# Assign a user as the editor of this submission
@scoobies assign @username as editor

# Remove the editor assigned to this submission
@scoobies remove editor

# Add a user to this issue's assignees list
@scoobies add assignee: @username

# Remove a user from this issue's assignees list
@scoobies remove assignee: @username

# Builds paper
@scoobies build paper

# Checks build status
@scoobies build status

# Check the references of the paper for missing DOIs
# Optionally, it can be run on a non-default branch 
@scoobies check references
@scoobies check references from custom-branch-name

# Label issue with: paper
@scoobies mark paper

# Label issue with: needs-review. Remove labels: unready, needs-more-review, pending-comment, ready
@scoobies mark needs review

# Label issue with: needs-more-review. Remove labels: unready, needs-review, pending-comment, ready
@scoobies mark needs more review

# Label issue with: pending-comment. Remove labels: unready, needs-review, needs-more-review, ready
@scoobies mark pending comment

# Label issue with: ready. Remove labels: unready, needs-review, needs-more-review, pending-comment
@scoobies mark ready

# Label issue with: unready. Remove labels: needs-review, needs-more-review, pending-comment, ready
@scoobies mark not ready

# Label issue with: does_not_build:server
@scoobies mark server fail

# Remove labels: does_not_build:server
@scoobies mark server success

```

kapoorlab commented:
@scoobies build paper

scoobies commented:
building paper...

scoobies commented:
{"data"=>{"info"=>"Build for paper kapoorlab-2021 scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

stargaser commented:
Hey @NickleDave, @davekits just a friendly reminder that the initial review deadline is next week.

NickleDave commented:
@stargaser thank you for the reminder

👋 hi @kapoorlab looking forward to reading and working with you on the review.

This is on my to-do list, will get it done by Wednesday night at the latest

kapoorlab commented:
@NickleDave Thank you so much, looking forward to your review :)

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+
+.. class:: abstract
+
+Live-cell imaging is a highly used technique to study cell migration and dynamics over time. Although many computational tools have been developed during the past years to automatically detect and track cells, they are optimized to detect cell nuclei with similar shapes and/or cells not clustering together. However, automated analysis of fluorescently membrane-labelled cells can be highly challen ...
```
I'd suggest rephrasing to make it clear the main problem you are addressing.

Perhaps change

> However, automated analysis of fluorescently membrane-labelled cells can be highly challenging 

to

> These existing tools are challenged when tracking fluorescently labeled membranes of cells, due to cells' irregular shape, ...

kapoorlab replied:
Hi there, I changed the abstract to: 

> These existing tools are challenged when tracking fluorescently labelled membranes of cells due to cell's irregular shape, variability in size and dynamic movement across Z planes making it difficult to detect and track them.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+Introduction
+------------
+
+Quantitative analysis to study morphogenesis requires an accurate image analysis workflow. Such a workflow entails volumetric (3D) imaging of cells via fluorescence microscopy, accurate detection and segmentation of cells followed by cell tracking and track analysis. Broadly speaking the task of segmentation can be separated into semantic segmentation (classifying pixels as background or pixels b ...
```
> Quantitative analysis to study morphogenesis requires an accurate image analysis workflow.

I might start with a simpler sentence that's less intimidating to non-experts (like me).

What is morphogenesis?
Doesn't any quantitative analysis require an accurate image analysis workflow?

Maybe say something like:
> Quantitative analysis of imaging data requires accuracy. For example, morphogenesis, [definition here], entails volumetric (3D) imaging of cells ...

kapoorlab replied:
I changed the statement to

> Quantitative analysis of imaging data requires accuracy. For example, morphogenesis, a process of shaping an organism by embryological process of cell diffrentiation, entails volumetric (3D) imaging of cells inside the tissue. To analyse the data an accurate image analysis workflow is required.

kapoorlab replied:
Actually maybe we can exclude mentioning morphogenesis here and just use cell migration instead and change the statement to this instead:

> Live-cell imaging is a highly used technique to study cell migration and dynamics over time. The image analysis workflow of volumetric (3D) imaging of cells via fluorescence microscopy starts with an accurate detection and segmentation of cells followed by cell tracking and track analysis.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+
+Quantitative analysis to study morphogenesis requires an accurate image analysis workflow. Such a workflow entails volumetric (3D) imaging of cells via fluorescence microscopy, accurate detection and segmentation of cells followed by cell tracking and track analysis. Broadly speaking the task of segmentation can be separated into semantic segmentation (classifying pixels as background or pixels b ...
+   
+Our dataset is comprised of epithelial cells of mouse embryonic mammary glands with membrane labelling. These cells are highly irregular in shape and have a low signal-to-noise ratio to obtain an accurate segmentation only based on the boundary information. Therefore, here we developed a segmentation package in python called VollSeg, as a method to perform automated segmentation of such cells. In ...
```
It's a bit confusing to start this paragraph talking about the dataset. How about something like this?
> To address the issues with existing segmentation algorithms just described, we developed Vollseg. In brief ... [rest of paragraph here] ... and U-Net networks. We benchmark our segmentation result on a challenging dataset comprised of epithelial cells of mouse embryonic mammary glands with membrane labelling. These cells are highly irregular in shape and have a low signal-to-noise ratio to obtain an accurate segmentation only based on the boundary information. Using this dataset, we obtain different metrics showing that our approach is able to obtain shape approximation for the overlapping cells that go beyond the star convex shape."

kapoorlab replied:
Changed this paragraph to 

> To address the issues with existing segmentation algorithms just described, we developed Vollseg. In brief we use Stardist in 3D to obtain a star convex shape approximation for the cells and extract the cell centroids from these polygons. We also train a 3D U-Net model to obtain a semantic segmentation map of the cells. We then perform a marker controlled watershed on the probability map of Stardist using the U-Net segmentation as a mask image to prevent the overflow of segmentation regions. To avoid the error of sub-optimal seeding we developed a seed pooling approach taking advantage of strength of both the Stardist and U-Net networks. We benchmark our segmentation result on a challenging dataset comprised of epithelial cells of mouse embryonic mammary glands with membrane labelling. These cells are highly irregular in shape and have a low signal-to-noise ratio to obtain an accurate segmentation only based on the boundary information. Using this dataset, we obtain different metrics showing that our approach is able to obtain shape approximation for the overlapping cells that go beyond the star convex shape.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+***************************
+
+
+We used fluorescent microscopy images of mouse embryonic mammary glands stabilized in an ex vivo culture previously collected in the laboratory of Dr. S. Fre at Institut Curie. All images were acquired with an inverted confocal laser scanning microscope (e.g. Zeiss LSM780/880) equipped with long-working distance objectives to acquire high-resolution 3D image stacks. We acquired images of pixel si ...
```
I'm not sure you need to provide a lengthy justification for image restoration here?
I think it's fine to just say:
> We acquired images of pixel size (22, 512, 512) with calibration of (3, 0.52, 0.52) micrometer. Lastly we performed unsupervised image denoising (cite) on our dataset, an algorithm we chose based on its performance compared to other methods (cite more)

kapoorlab replied:
Changed this paragraph to 

> We acquired images of pixel size (22, 512, 512) with calibration of (3, 0.52, 0.52) micrometer. 
The quality at which these images are acquired is determined by the spatial resolution of the used optical device, desired temporal resolution, duration of the experiment and depth of the acquired Z-stacks. We perform unsupervised image denoising :cite:`krull2019` on our dataset, an algorithm we chose based on its performance compared to other methods :cite:`Richardson72`, :cite:`Lucy74` and due to lack of having training data needed for supervised learning algorithms :cite:`Weigert2017`.

kapoorlab replied:
I removed the mention of supervised learning algorithm and change this paragraph to say the following:

>  We acquired images of pixel size (22, 512, 512) with calibration of (3, 0.52, 0.52) micrometer. 
The quality at which these images are acquired is determined by the spatial resolution of the used optical device, desired temporal resolution, duration of the experiment and depth of the acquired Z-stacks. We perform unsupervised image denoising :cite:`krull2019` on our dataset, an algorithm we chose based on its performance compared to other methods :cite:`Richardson72`, :cite:`Lucy74`.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+Segmentation
+**************
+
+Post-restoration of the 3D images, we developed a method to perform the segmentation of the cells using deep learning techniques, as it was shown by Rasse et al. that conventional computer vision and machine learning based techniques alone will almost always lead to sub-optimal segmentation :cite:`Rasse2020`. We created a training dataset with hand drawn segmentation of 14 Z-stacks. We performed  ...
```
I would move this language to the introduction, to the place where you originally cite this paper.
> as it was shown by Rasse et al. that conventional computer vision and machine learning based techniques alone will almost always lead to sub-optimal segmentation :cite:`Rasse2020`

It's more explicit than the passing mention of the paper you have there now and helps motivate your development of the method.


I'm not sure you need higher-level comments like this now that you're already describing the method :slightly_smiling_face:

NickleDave replied:
> (computed using skimage)

I would spell out the entire name "scikit-image" and include a citation here, to the paper and to the DOI of the specific version of `skimage` that you used

kapoorlab replied:
Moved the statement to introduction appending it to the original statement there to now say

> It was shown in :cite:`Rasse2020` that conventional computer vision and machine learning based techniques alone will almost always lead to sub-optimal segmentation and that methods based on deep learning have improved the accuracy of segmentation for natural and biomedical images alike.

Added citation of scikit-image version 0.18x, am not sure if I found the right DOI for it though, I cite the paper mentioned here: https://scikit-image.org/

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+Segmentation
+**************
+
+Post-restoration of the 3D images, we developed a method to perform the segmentation of the cells using deep learning techniques, as it was shown by Rasse et al. that conventional computer vision and machine learning based techniques alone will almost always lead to sub-optimal segmentation :cite:`Rasse2020`. We created a training dataset with hand drawn segmentation of 14 Z-stacks. We performed  ...
```
re: section names, I'd suggest changing the section name "Segmentation" to something like "Dataset Preparation". At least at the start of the paragraph here it sounds like you are describing steps you took to prepare the data for training and evaluating your algorithm. Not sure if there's a more precise term to use -- I thought of "munging" or "canonicolization" :/

NickleDave replied:
Before this sentence

> Stardist predicts object instances based on probability threshold and non maximal suppression threshold to merge overlapping predictions. These parameters can be automatically determined using the optimize threshold program that we provide with the segmentation package.

I'd suggest adding a new section, e.g. "Parameter Tuning" or "Parameter Selection"

kapoorlab replied:
Since the section before Segmentation is already called Preparation of  the dataset, I just merged the two sections into on and before making the statement about stardist I added a section called Parameter Tuning.

kapoorlab replied:
Shall we call this Parameter Setting instead? since its not a manual choice of parameters and we are determining it via computing iou scores of the prediction and the validation dataset. Then I keep the title of segmentation and the text then looks as



> Parameter Setting
*****************
Stardist predicts object instances based on probability threshold and non maximal suppression threshold to merge overlapping predictions. These parameters can be automatically determined using the optimize threshold program that we provide with the segmentation package. Higher values of the probability threshold yield fewer object instances, but avoids false positives. Higher values of the overlap threshold would lead to oversegmentation. We used 32 Z-stacks to determine the optimal parameters of probability threshold of 0.76 and non maximal suppression threshold of 0.3. 

Segmentation
*****************
As illustrated in Figure :ref:`algorithm`, we first obtain the centroids of the star convex approximated cell shapes and create a seed pool with these centroid locations.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+    
+    
+    
+Accuracy of segmentation results is assesed by comparing the obtained labels to the gold standard ground truth (GT) labels. Most commonly used metric is to compute intersection over union (IOU) score between the predicted and the GT label image.
```
I'd suggest making this sentence the start of a new section named something like "Performance Metrics"

NickleDave replied:
Perhaps remove the phrase "gold standard"

> Accuracy of segmentation results is assesed by comparing the obtained labels to the gold standard ground truth (GT) labels.

"ground truth" has a technical meaning that is appropriate here but to me "gold standard" is more of a comment on why a certain approach is used. In this context I think it's clear you'd want to use the "gold standard" approach.

NickleDave replied:
> Most commonly used metric is to compute intersection over union (IOU) score between the predicted and the GT label image.

Insert "The" before "most"

kapoorlab replied:
Added the section performance metrics and remove the gold standard words as it is really the ground truth labels that we want to compare the segmentation results with. So in that case I thought it was most appropriate to remove the gold standard and just call it ground truth labels. 

Inserted The before most.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+:math:`AP_\tau= \frac{TP_\tau}{TP_\tau+ FP_\tau + FN_\tau}`.  
+To evaluate the accuracy of our method in resolving the shape of the cells we compute the mean squared error (MSE) and structural similarity index measurment (SSIM) between the GT and obtained segmentation images post-binarization operation on the obtained instance segmentation maps. MSE shows a low score if the image is structurally closer to GT. SSIM score is higher if the two images are struct ...
+    
+The software package we provide comes with training and prediction notebooks for training the base U-Net and Stardist networks on your own dataset. We provide jupyter notebooks to do so on local GPU servers and also on Google Colab.
```
I think you intend this sentence ("The software package...") to be the start of a new sub-section within "Methods", where you walk through the process in detail, before you get to the "Results" section?

If so, I would add a new sub-section title here, e.g. "Walkthrough" (or if you prefer "Protocol" like you say in the abstract) and then making the following section headers one level lower, i.e. "Walkthrough" will be a level two heading and "Network Training..." would be level 3

```rst
Methods
-----------

...

Walkthrough
**************

The software package we provide comes with training and prediction notebooks for training the base U-Net and Stardist networks on your own dataset. We provide jupyter notebooks to do so on local GPU servers and also on Google Colab.

======================
Network Training
======================
```

kapoorlab replied:
Yeah, I called it protocol instead of walkthrough because it is commonly used terminology in biological sciences.
I redid the sectioning as you mentioned with the levels as you said.

kapoorlab replied:
I changed protocol to Detailed Procedure to distinguish it from a more technical usage of protocol in the field of biological sciences.

NickleDave commented on papers/varun_kapoor/00_varunclaudia.rst:
```diff
@@ -0,0 +1,354 @@
+
+Post-restoration of the 3D images, we developed a method to perform the segmentation of the cells using deep learning techniques, as it was shown by Rasse et al. that conventional computer vision and machine learning based techniques alone will almost always lead to sub-optimal segmentation :cite:`Rasse2020`. We created a training dataset with hand drawn segmentation of 14 Z-stacks. We performed  ...
+
+.. figure:: Figures/Seg_pipe.png
```
Consider moving this figure to the first page if possible and referencing it in the second paragraph when you describe what is unique about your method

I think it would help make your contribution clearer to a reader

kapoorlab replied:
I was able to move the figure to the second page in introduction and make a reference to it in the second paragraph where I describe the method. Somehow the compiler does not place the figure on page 1, would that be acceptable? I will just commit my new paper and ask scoobies to build it to see if everything is in order. Looking forward to hearing from you :)

NickleDave reviewed (changes requested):
Hi @kapoorlab thank you for your submission. I enjoyed reading and learning about your proposed algorithm and the software you've developed.

My initial assessment is that this is a very solid contribution. I think we can work together to have the paper in a place where I can recommend including it in the proceedings by the final deadline.

I've begun adding comments throughout. I need to finish going through, but I wanted to get the process started, to give you a chance to address my suggested revisions (and so I actually meet the initial deadline :angel: )

deniederhut commented:
@scoobies mark needs more review

kapoorlab commented:
@NickleDave Thank you so much. I will work on your comments over the weekend and make all the necessary changes you asked for by Sunday night :). Thank you so much for making our contribution nicer.

kapoorlab committed:
abstract and introduction

Commit:
Merge branch '2021' into 2021

kapoorlab committed:
introduction paragraph change

kapoorlab committed:
Merge branch '2021' of https://github.com/kapoorlab/scipy_proceedings into 2021

kapoorlab committed:
image restoration

kapoorlab committed:
scikit-image reference

kapoorlab committed:
add parameter tuning section

kapoorlab committed:
remove gold standard

kapoorlab committed:
Protocol sectioning

kapoorlab committed:
move figure up

kapoorlab committed:
move figure to page 2

kapoorlab commented:
@scoobies build paper

scoobies commented:
building paper...

scoobies commented:
{"data"=>{"info"=>"Build for paper kapoorlab-2021 scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

kapoorlab commented:
@scoobies build paper

scoobies commented:
building paper...

scoobies commented:
{"data"=>{"info"=>"Build for paper kapoorlab-2021 scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

kapoorlab committed:
format text

kapoorlab committed:
format text

kapoorlab committed:
re-organie text

kapoorlab commented:
@scoobies build paper

scoobies commented:
building paper...

scoobies commented:
{"data"=>{"info"=>"Build for paper kapoorlab-2021 scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

kapoorlab committed:
adding poster

kapoorlab commented:
@deniederhut I added our poster in the presentation folder as a pdf file.

deniederhut commented:
@kapoorlab to give everyone more time to respond to their reviews, we've pushed back the deadline for authors' final revisions to July 1st

stargaser commented:
@NickleDave thank you for reviewing this paper. Do the revisions made by @kapoorlab address the issues you raised in your review?

stargaser commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.7717/peerj.453 is OK
- 10.1038/nmeth.1228 is OK
- 10.1364/JOSA.62.000055 is OK
- 10.1101/236463 is OK
- 10.1101/2020.01.17.910562 is OK
- 10.1038/s41592-019-0582-9 is OK
- 10.1038/s41592-018-0216-7 is OK
- 10.3389/fbioe.2020.558880 is OK
- 10.1101/2020.09.10.276980 is OK
- 10.1101/gad.322818.118 is OK
- 10.1109/WACV45572.2020.9093435 is OK
- 10.1007/978-3-030-00934-2_30 is OK

MISSING DOIs

- 10.1086/111605 may be a valid DOI for title: An iterative technique for the rectification of observed distributions
- 10.1109/isbi.2019.8759242 may be a valid DOI for title: CNN-based Preprocessing to Optimize Watershed-based Cell Segmentation in 3D Confocal Microscopy Images
- 10.1201/9781482277234-12 may be a valid DOI for title: The Morphological Approach to Segmentation: The Watershed Transformation
- 10.1109/cvpr.2019.00223 may be a valid DOI for title: Noise2void-learning denoising from single noisy images

INVALID DOIs

- https://doi.org/10.1002/nav.3800020109 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.ymeth.2016.09.016 is INVALID because of 'https://doi.org/' prefix
```

stargaser commented:
@kapoorlab here are some items from an editorial check:

- not overruning margins on code blocks: good
- DOIs in references: mostly good, a few may need to be added or revised, please see previous comment for the DOI check
- Not modifying files outside the papers directory: this definitely needs attention as there are many top-level files and files in the `publisher` directory that are modified in this pull request. Those files must be removed before we can merge and publish this contribution.

kapoorlab committed:
publisher dir

kapoorlab committed:
modify dois

kapoorlab commented:
@stargaser thanks for checks. I corrected the doi's and removed the conflicting files from the publisher folder. I pushed my changes. Does everything look okay now?

stargaser commented on papers/varun_kapoor/vkc.bib:
```diff
@@ -0,0 +1,321 @@
+ pages = {e453},
+ journal = {PeerJ},
+ issn = {2167-8359},
+ url = {https://doi.org/10.7717/peerj.453},
```
here and elsewhere in the references, the `url` item is not needed when the `doi` item is present. The `doi` item generates a URL in the paper PDF.

kapoorlab replied:
I removed the urls's from the citations that had doi's included. Making a push

stargaser commented:
Thanks @kapoorlab for updating the DOIs. In some of the references, the `url` item with the same information can be removed as it makes an extra link in the paper PDF.

Thanks also for removing the `publisher` subdirectory. There are still some top-level files in the pull request that need removal before merging: `README.md`, `requirements.txt`, `review_criteria.md`, `scipy.proc.json`.

I see you've included the poster PDF in this pull request. I need to check with @deniederhut and the other co-chairs about that.

deniederhut commented:
> I see you've included the poster PDF in this pull request. 

As long as the required metadata is there, this is okay

deniederhut commented:
@scoobies mark pending comment

kapoorlab committed:
remove urls with doi in list

kapoorlab committed:
remove more files

kapoorlab commented:
@stargaser I removed the other files form the top level directory and updated the bib file. Is the pull request compatible now?

kapoorlab commented:
@deniederhut I received an email about the poster as well where I was supposed to send the link to the poster, I put the link from my github, where the poster is. Would that be fine of shall I do it some other way?

stargaser commented:
My apologies @kapoorlab, I should have said that the files in the top-level directory should not differ from those on the scipy_proceedings `2021` branch so that they do not appear in the "Files Changed" tab here on the pull request.

If your remote for the `scipy-conference/scipy_proceedings` repository is named `upstream`, then I think you can get these with
```
git fetch upstream
git checkout upstream/2021 README.md
```
and so on for those top-level files; then you can commit them.

Regarding the poster itself, adding it to this repository is for upload to Zenodo and including a link to it in the SciPy proceedings. I don't know what the instructions are for displaying your poster at the conference.

kapoorlab committed:
top level dir

NickleDave commented:
> @NickleDave thank you for reviewing this paper. Do the revisions made by @kapoorlab address the issues you raised in your review?

Hi @stargaser sorry I missed this.

Yes, I have verified that the revisions made by @kapoorlab address the issues I raised in review. Thank you @kapoorlab !
I hope my feedback was helpful.

kapoorlab commented:
@NickleDave Thank you for reviewing the paper, yes your review was helpful in structuring it correctly. Thank you for approving the changes. @stargaser I did the git fetches and got it all in sync. Poster instructions came to in separate emails with a form to fill where you need to provide the poster link, which is why I put the poster in here and then we all will present it in gather town.

deniederhut commented:
> I put the link from my github, where the poster is. Would that be fine of shall I do it some other way?

That should be okay? You can check in with Madicken and Matt if you aren't sure.

stargaser commented:
@kapoolab thank you for fetching the top-level `README.md` and committing it. Would you please do the same for top-level `review_criteria.md`, `requirements.txt` and `scipy_proc.json`? These files still appear in the "Files changed" tab.

stargaser commented:
@kapoorlab  thank you for fetching the top-level README.md and committing it. Would you please do the same for top-level review_criteria.md, requirements.txt and scipy_proc.json? These files still appear in the "Files changed" tab.

kapoorlab committed:
Delete requirements.txt

kapoorlab committed:
top level changes

kapoorlab commented:
@stargaser thank you for pointing this out now I think now it should be fine. My pull request should be mergable with the main branch. Let me know if this is not the case.

deniederhut commented:
Hi there! It looks your poster pdf is missing some metadata (in the presentations directory). We don't need to fix this for the paper, but we'll need to fix it to publish the rendered poster. You can see instructions for how to do that here: https://github.com/scipy-conference/scipy_proceedings#instructions-for-slides

kapoorlab committed:
add poster metadata

kapoorlab commented:
@deniederhut I added the info.json file in the poster directory. It should be okay now. Many thanks.

stargaser commented:
@scoobies mark ready
